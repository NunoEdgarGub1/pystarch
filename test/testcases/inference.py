
x = 5 if random() > 0.5 else None
y = 2 if random() > 0.2 else None

if x is None or y is None:
    z = x + y
else:
    z = x * y
