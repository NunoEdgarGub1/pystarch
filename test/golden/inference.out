x Maybe(Num)
y Maybe(Num)
z Num

testcases/inference.py:2 undefined "random" (random)
testcases/inference.py:3 undefined "random" (random)
testcases/inference.py:6 type-error "x" (Maybe(Num) vs Num)
testcases/inference.py:6 type-error "y" (Maybe(Num) vs Num)
